### PR TITLE
Invariant Improvements: Add settle invariant scenario

### PR DESCRIPTION
**Pull request by prat-gpt**
# Description of change
## High level
* Add settle invariant scenario where anyone can come and repay borrowers loan and loan is always settleable after 72 hours of kick. See: https://www.notion.so/worlds-best-crypto-lending-platform/Settle-Invariants-1318afdbf9b64dd28cb3225664c66130

### CODE REVIEW

File: tests/forge/invariants/ERC20Pool/handlers/SettleERC20PoolHandler.sol [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+        if (block.timestamp < kickTime + 72 hours) {
+            skip(kickTime + 73 hours - block.timestamp);
+        }
+
+        // skip time to make auction settleable
+        if (block.timestamp < kickTime + 72 hours) {
+            skip(kickTime + 73 hours - block.timestamp);
+        }
+
+        changePrank(payer);
```
Comment (EdNoepel) [inline tests/forge/invariants/ERC20Pool/handlers/SettleERC20PoolHandler.sol:131]: Recommend moving the `changePrank` into the utility method below.

**Review by EdNoepel — APPROVED** [review]

File: tests/forge/invariants/ERC20Pool/handlers/SettleERC20PoolHandler.sol (hunk 1 above)
Comment (prat-gpt) [reply tests/forge/invariants/ERC20Pool/handlers/SettleERC20PoolHandler.sol:131]: I prefer the `changePrank` here to maintain consistency with rest of the code. I do all the skips, bounds and changePranks only in external methods.

**Review by MikeHathaway — APPROVED** [review]
LGTM - would be good to do this for ERC721 pools as well

**Comment by ith-harvey** [comment]
> LGTM - would be good to do this for ERC721 pools as well

I second this, would be worthwhile to do for ERC721 pools

**Review by ith-harvey — APPROVED** [review]

**Pull request merged**